app/src/main/java/org/simple/clinic/home/overdue/OverdueListAdapter.kt

```diff
@@ -20,7 +20,10 @@ import java.util.UUID
 
 
 class OverdueListAdapter(
-    private val phoneCallClickStream: PublishSubject<CallPatientClicked>
+    private val phoneCallClickStream: PublishSubject<CallPatientClicked>,
+    private val agreedToVisitClickStream: PublishSubject<AgreedToVisitClicked>,
+    private val remindLaterClickStream: PublishSubject<RemindToCallLaterClicked>,
+    private val removeFromListClickStream: PublishSubject<RemoveFromListClicked>
```

Comment:
Tip: We could expose these streams from the adapter instead of receiving it in the constructor. This way, this adapter can also be injected by DI.

In the calling side, you could then do:

```kotlin
Observable.mergeArray(screenCreates(), callPermissionChanges(), adapter.uiEvents)
  .compose(controller)
  ...
```

Reply:
I like this idea. Nice.